pkg/monitor/monitorapi/types.go

```diff
@@ -197,6 +197,29 @@ func IsInNamespaces(namespaces sets.String) EventIntervalMatchesFunc {
 	}
 }
 
+// ContainsAllParts ensures that all listed key match at least one of the values.
+func ContainsAllParts(matchers map[string][]string) EventIntervalMatchesFunc {
+	return func(eventInterval EventInterval) bool {
+		actualParts := LocatorParts(eventInterval.Locator)
+		for key, possibleValues := range matchers {
+			actualValue := actualParts[key]
+
+			found := false
+			for _, possibleValue := range possibleValues {
+				if actualValue == possibleValue {
```

Comment:
How about `strings.Contains(actualValue, possibleValue)`? this way I can filter for all pods of a certain pattern or maybe make it so we can pass a regex?

Reply:
> How about strings.Contains(actualValue, possibleValue)? this way I can filter for all pods of a certain pattern or maybe make it so we can pass a regex?

A good idea.  Could you make it work with regexes and negative matching as a followup to this?

Reply:
sure.  The last thing my script has that yours doesn't is the negative match.  I can do that on a followup PR.

Reply:
@deads2k I'm ok with this as-is and I can have a PR up shortly after this lands for the regex and negative match.